lua/entities/gmod_wire_expression2/core/custom/tardis.lua

```diff
@@ -1,575 +1,207 @@
-		local success=ent:ToggleLocked()
-		if success then
-			return 1
-		else
-			return 0
-		end
-	end
-	return 0
-end
-
-local function TARDIS_Physlock(data,ent)
-	if ent and IsValid(ent) and CheckPP(data.player,ent) then
-		if not (ent:GetClass()=="sent_tardis") then return 0 end
-		local success=ent:TogglePhysLock()
-		if success then
-			return 1
-		else
-			return 0
-		end
-	end
-	return 0
-end
-
-local function TARDIS_Power(data,ent)
-	if ent and IsValid(ent) and CheckPP(data.player,ent) then
-		if not (ent:GetClass()=="sent_tardis") then return 0 end
-		local success=ent:TogglePower()
-		if success then
-			return 1
-		else
-			return 0
-		end
-	end
-	return 0
-end
-
-local function TARDIS_Isomorph(data,ent)
-	if ent and IsValid(ent) and CheckPP(data.player,ent) then
-		if not (ent:GetClass()=="sent_tardis") or not IsValid(data.player) then return 0 end
-		local success=ent:IsomorphicToggle(data.player)
-		if success then
-			return 1
-		else
-			return 0
-		end
-	end
-	return 0
-end
-
-local function TARDIS_Longflight(data,ent)
-	if ent and IsValid(ent) and CheckPP(data.player,ent) then
-		if not (ent:GetClass()=="sent_tardis") then return 0 end
-		local success=ent:ToggleLongFlight()
-		if success then
-			return 1
-		else
-			return 0
-		end
-	end
-	return 0
-end
-
-local function TARDIS_Materialise(data,ent)
-	if ent and IsValid(ent) and CheckPP(data.player,ent) then
-		if not (ent:GetClass()=="sent_tardis") then return 0 end
-		local success=ent:LongReappear()
-		if success then
-			return 1
-		else
-			return 0
-		end
-	end
-	return 0
-end
-
-local function TARDIS_Selfrepair(data,ent)
-	if ent and IsValid(ent) and CheckPP(data.player,ent) then
-		if not (ent:GetClass()=="sent_tardis") then return 0 end
-		local success=ent:ToggleRepair()
-		if success then
-			return 1
-		else
-			return 0
-		end
-	end
-	return 0
-end
-
-local function TARDIS_Track(data,ent,trackent)
-	if ent and IsValid(ent) and CheckPP(data.player,ent) then
-		if not (ent:GetClass()=="sent_tardis") then return 0 end
-		local success=ent:SetTrackingEnt(trackent)
-		if success then
-			return 1
-		else
-			return 0
-		end
-	end
-	return 0
-end
-
-local function TARDIS_Spinmode(data,ent,spinmode)
-	if ent and IsValid(ent) and CheckPP(data.player,ent) then
-		if not (ent:GetClass()=="sent_tardis") then return 0 end
-		ent:SetSpinMode(spinmode)
-		return ent.spinmode
-	end
-	return 0
-end
-
-local function TARDIS_SetDestination(data,ent,pos,ang)
-	if ent and IsValid(ent) and CheckPP(data.player,ent) then
-		if not (ent:GetClass()=="sent_tardis") then return 0 end
-		local pos=Vector(pos[1], pos[2], pos[3])
-		if ang then ang=Angle(ang[1], ang[2], ang[3]) end
-		if ent.invortex then
-			ent:SetDestination(pos,ang)
-			return 1
-		else
-			return 0
-		end
+local function HandleE2(ent, name, ...)
+	if IsValid(getTardis(ent)) then
+		return ent:HandleE2(name, ...)
 	else
-		return 0
+		error("Can't call TARDIS functions on something other than a TARDIS.")
```

Comment:
Can you make sure that this doesn't crash wiremod if this happens? Should just return an E2 error

Reply:
It doesn't crash wiremod, only the expression. I'm not sure how to return E2 errors

Reply:
Done now, adding a zero as the level made the error be thrown as an e2 error, but it doesn't show lines and stuff.